src/sentry/api/endpoints/organization_releases.py

```diff
@@ -535,7 +535,7 @@ def post(self, request: Request, organization) -> Response:
                     ]
                 scope.set_tag("has_refs", bool(refs))
                 if refs:
-                    if not request.user.is_authenticated:
+                    if not request.user.is_authenticated and not request.auth:
```

Comment:
Without a user in the request project plugins may not be able to fetch commits. The two project plugins that implement commit fetching have organization integrations that customers could switch to. Might be worth mentioning the incompatibility with project plugins in the docs for org auth tokens 🤷